config/v1/types_proxy.go

```diff
@@ -34,6 +34,10 @@ type ProxySpec struct {
 	// Empty means unset and will not result in an env var.
 	// +optional
 	NoProxy string `json:"noProxy,omitempty"`
+
+	// readinessEndpoints is a list of endpoints used to verify readiness of the proxy.
+	// +optional
+	ReadinessEndpoints []string `json:"readinessEndpoints,omitempty"`
```

Comment:
one (minorish) concern i have w/ this is that this struct is already used as part of other config objects(embedded in them) which will not currently respect the readinessendpoints field.  So this godoc may need some clarification.  (or some apis need to be refactored such that existing embedders of the ProxySpec struct do not inherit this new field.)

/cc @adambkaplan

Reply:
> or some apis need to be refactored such that existing embedders of the ProxySpec struct do not inherit this new field.

@bparees I think refactoring is the most appropriate thing to do here.

Reply:
@bparees it sounds like `BuildDefaults` is the only embedder of `ProxySpec`. @adambkaplan will create a new struct (i.e. `ProxyInfo`) based on `ProxySpec` w/o the unneeded fields and  refactor [BuildDefaults](https://github.com/openshift/api/blob/master/config/v1/types_build.go#L43-L50) to use it.

Reply:
if it works for @adambkaplan it works for me.

Reply:
@bparees I discussed with @danehans on slack - this works.